src/main.js

```diff
@@ -0,0 +1,159 @@
+import React from 'react';
+import { render as reactDomRender } from 'react-dom'; 
+import superagent from 'superagent';
+import './style/main.scss';
+
+const apiUrl = 'http://www.reddit.com/r';
+// /${searchFormBoard}.json?limit=${searchFormLimit}
+
+class SearchForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      redditBoard: '',
+      limit: '',
+    };
+
+    this.handleChange = this.handleChange.bind(this);
+    this.handleLimitChange = this.handleLimitChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleChange(event) {
+    this.setState({ redditBoard: event.target.value });
+  }
+
+  handleLimitChange(event) {
+    this.setState({ limit: event.target.value });
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+    this.props.redditBoardSelect(this.state.redditBoard);
+  }
+
+  render() {
+    return (
+      <form onSubmit={this.handleSubmit}>
+        <input 
+          type="text"
+          name="boardName"
+          placeholder="Search for a board"
+          value={this.state.redditBoard}
+          onChange={this.handleChange}
+        />
+      </form>
+    );
+  }
+}
+
+// class SearchResultList extends React.Component {
+//   constructor(props) {
+//     super(props);
+
+//     this.renderRedditBoardList = this.renderRedditBoardList.bind(this);
+//   }
+
+//   renderRedditBoardList(board) {
+//     this.setState({ renderRedditBoardList: board.map(item, index) });
+//   }
+
+//   render() {
+//     return (
+//       <ul>
+//         <li key={index}>
+//           <a href={item.url}>{item.title}<p>{item.ups}</p></a>
+//         </li>
+//       </ul>
+//     );
+//   }
+// }
+
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      redditBoardSelected: null,
+      redditBoardError: null,
+    };
+
+    this.redditBoardSelect = this.redditBoardSelect.bind(this);
+  }
+
+  redditBoardSelect(board) {
+    if (!`${apiUrl}/penguins.json?limit=5`) {
+      this.setState({
+        redditBoardSelected: null,
+        redditBoardError: name,
+      });
+    } else {
+      return superagent.get(`${apiUrl}/penguins.json?limit=5`) 
```

Comment:
make your superagent.get request dynamic with the parameters as template literal values in the api call.
```javascript
return superagent.get(`${apiUrl}/${board}.json?limit=${limit}`)
```